From 6a0f482213d3a439fd3382f6b624b32f98a02d25 Mon Sep 17 00:00:00 2001
From: Tim Cera <tim@cerazone.net>
Date: Thu, 4 Jan 2024 17:35:44 -0500
Subject: [PATCH] Windows does not have a separate math library

---
 src/CMakeLists.txt | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 41e11f0..73e38d6 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -175,12 +175,14 @@ CONFIGURE_FILE("${CMAKE_CURRENT_SOURCE_DIR}/grib2.h.in" "${CMAKE_CURRENT_BINARY_
 target_include_directories(${lib_name}_objlib PUBLIC "$<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR};${CMAKE_CURRENT_SOURCE_DIR}>"
                                                      $<INSTALL_INTERFACE:${CMAKE_INSTALL_PREFIX}/include>)
 
-target_link_libraries(${lib_name}_objlib INTERFACE m)
-if(BUILD_SHARED_LIBS)
-  target_link_libraries(${lib_name}_shared INTERFACE m)
-endif()
-if(BUILD_STATIC_LIBS)
-  target_link_libraries(${lib_name}_static INTERFACE m)
+if (NOT WIN32)
+  target_link_libraries(${lib_name}_objlib INTERFACE m)
+  if(BUILD_SHARED_LIBS)
+    target_link_libraries(${lib_name}_shared INTERFACE m)
+  endif()
+  if(BUILD_STATIC_LIBS)
+    target_link_libraries(${lib_name}_static INTERFACE m)
+  endif()
 endif()
 if (WIN32)
   target_link_libraries(${lib_name}_objlib INTERFACE ws2_32)
-- 
2.34.1

