--- g2ccsv.c	2024-10-30 13:06:17
+++ g2ccsv.c_FIX	2025-02-27 10:45:23
@@ -9,6 +9,8 @@
 #include <stdlib.h>
 #include <string.h>
 
+#define MAX_LINE_SIZE 500
+
 /** Contains the parsed CSV document. */
 FILE *doc;
 
@@ -188,11 +190,10 @@
 int
 g2c_csv_init()
 {
-    const int max_line_size = 500;
     const int num_columns = 9;
     int i;
     char *buf, *tmp, *key;
-    char line[max_line_size];
+    char line[MAX_LINE_SIZE];
     G2C_CODE_TABLE_T *my_table = NULL;
     G2C_CODE_ENTRY_T *new_entry = NULL;
 
@@ -206,11 +207,11 @@
         return G2C_ECSV;
 
     /* Skip header line */
-    buf = fgets(line, max_line_size, doc);
+    buf = fgets(line, MAX_LINE_SIZE, doc);
 
     /* Go through the document and save table data.
      * Each line is a table of codes. */
-    while ((buf = fgets(line, max_line_size, doc)) != NULL)
+    while ((buf = fgets(line, MAX_LINE_SIZE, doc)) != NULL)
     {
         i = 0;
         while (buf != NULL && i < num_columns)
